Error when installing extension from OCI image

The macadam.js library needs to be in the node_modules and be built as CommonJS

depends on https://github.com/crc-org/macadam.js/issues/18

```
Error: Cannot find module '@crc-org/macadam.js/package.json' Require stack: - /Users/phmartin/.local/share/containers/podman-desktop/plugins/quayiophmartinrhel/dist/extension.js - /Users/phmartin/Documents/gits/podman-desktop/packages/main/dist/index.cjs
Stack trace:
Error: Cannot find module '@crc-org/macadam.js/package.json'
Require stack:
- /Users/phmartin/.local/share/containers/podman-desktop/plugins/quayiophmartinrhel/dist/extension.js
- /Users/phmartin/Documents/gits/podman-desktop/packages/main/dist/index.cjs
    at Module._resolveFilename (node:internal/modules/cjs/loader:1232:15)
    at s._resolveFilename (node:electron/js2c/browser_init:2:129200)
    at resolve (node:internal/modules/helpers:146:19)
    at Macadam.findMacadamPath (/Users/phmartin/.local/share/containers/podman-desktop/plugins/quayiophmartinrhel/dist/extension.js:107:47)
    at Macadam._init (/Users/phmartin/.local/share/containers/podman-desktop/plugins/quayiophmartinrhel/dist/extension.js:85:40)
    at Macadam.init (/Users/phmartin/.local/share/containers/podman-desktop/plugins/quayiophmartinrhel/dist/extension.js:79:21)
    at activate (/Users/phmartin/.local/share/containers/podman-desktop/plugins/quayiophmartinrhel/dist/extension.js:843:25)
    at ExtensionLoader.activateExtension (/Users/phmartin/Documents/gits/podman-desktop/packages/main/dist/index.cjs:9358:59)
    at ExtensionLoader.loadExtension (/Users/phmartin/Documents/gits/podman-desktop/packages/main/dist/index.cjs:8651:20)
    at async ExtensionLoader.loadExtensions (/Users/phmartin/Documents/gits/podman-desktop/packages/main/dist/index.cjs:8461:7)

```

--- a/build/Containerfile
+++ b/build/Containerfile
@@ -23,7 +23,12 @@ RUN pnpm --frozen-lockfile install && \
     pnpm build
 
 RUN echo "node-linker=hoisted" >> /opt/app-root/src/dist/.npmrc
-RUN pnpm --dir /opt/app-root/src/dist add ssh2 @crc-org/macadam.js@0.0.1-202504180956-af4279b
+RUN eval 'dep_version() { pnpm list -P $1 --json --depth 0 |jq -r ".[0].dependencies.\"$1\".version"; };' && \
+        MACADAM_VERSION=$(dep_version "@crc-org/macadam.js") && \
+        SSH2_VERSION=$(dep_version "ssh2") && \
+        echo adding @crc-org/macadam.js version ${MACADAM_VERSION} && \
+        echo adding ssh2 version ${SSH2_VERSION} && \
+        pnpm --dir /opt/app-root/src/dist add ssh2@${SSH2_VERSION} @crc-org/macadam.js@${MACADAM_VERSION}
 
 FROM scratch
 

--- a/build/Containerfile
+++ b/build/Containerfile
@@ -23,7 +23,7 @@ RUN pnpm --frozen-lockfile install && \
     pnpm build
 
 RUN echo "node-linker=hoisted" >> /opt/app-root/src/dist/.npmrc
-RUN pnpm --dir /opt/app-root/src/dist add ssh2
+RUN pnpm --dir /opt/app-root/src/dist add ssh2 @crc-org/macadam.js@0.0.1-202504180956-af4279b
 
 FROM scratch
 

--- a/build/Containerfile.builder
+++ b/build/Containerfile.builder
@@ -17,6 +17,10 @@
 
 FROM registry.access.redhat.com/ubi9/nodejs-22
 
+USER root
+RUN dnf install -y jq
+USER default
+
 COPY package.json .
 COPY pnpm-lock.yaml .
 COPY .npmrc .

